crates/walrus-core/src/encoding/quilt_encoding.rs

```diff
@@ -2425,17 +2445,49 @@ mod tests {
         assert_eq!(borrowed_with_tags, owned_with_tags);
 
         // Different content should not be equal
-        let different = QuiltStoreBlob::new(&[1, 2, 3, 4, 6], "test-blob");
+        let different = QuiltStoreBlob::new(&[1, 2, 3, 4, 6], "test-blob")?;
         assert_ne!(borrowed, different);
 
         // Different identifier should not be equal
-        let different_id = QuiltStoreBlob::new(&data, "different-id");
+        let different_id = QuiltStoreBlob::new(&data, "different-id")?;
         assert_ne!(borrowed, different_id);
 
         // Different tags should not be equal
         let mut different_tags = BTreeMap::new();
         different_tags.insert("key1".to_string(), "different-value".to_string());
         let borrowed_different_tags = borrowed.with_tags(different_tags);
         assert_ne!(borrowed_with_tags, borrowed_different_tags);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_identifier_validation() {
+        // Valid identifiers.
+        assert!(QuiltStoreBlob::is_valid_identifier("test"));
+        assert!(QuiltStoreBlob::is_valid_identifier("test123"));
+        assert!(QuiltStoreBlob::is_valid_identifier("123test"));
+        assert!(QuiltStoreBlob::is_valid_identifier("a"));
+        assert!(QuiltStoreBlob::is_valid_identifier("A123"));
+
+        // Invalid identifiers - doesn't start with alphanumeric.
+        assert!(!QuiltStoreBlob::is_valid_identifier("_test"));
+        assert!(!QuiltStoreBlob::is_valid_identifier("-test"));
+        assert!(!QuiltStoreBlob::is_valid_identifier(" test"));
+        assert!(!QuiltStoreBlob::is_valid_identifier(".test"));
+        assert!(!QuiltStoreBlob::is_valid_identifier(""));
+
+        // Invalid identifiers - has trailing whitespace.
+        assert!(!QuiltStoreBlob::is_valid_identifier("test "));
+        assert!(!QuiltStoreBlob::is_valid_identifier("test\t"));
+        assert!(!QuiltStoreBlob::is_valid_identifier("test\n"));
+        assert!(!QuiltStoreBlob::is_valid_identifier("test\r"));
+        assert!(!QuiltStoreBlob::is_valid_identifier("test   "));
+
+        // Valid identifiers with spaces/special chars in the middle
+        assert!(QuiltStoreBlob::is_valid_identifier("test blob"));
+        assert!(QuiltStoreBlob::is_valid_identifier("test-blob"));
+        assert!(QuiltStoreBlob::is_valid_identifier("test_blob"));
+        assert!(QuiltStoreBlob::is_valid_identifier("test.blob"));
     }
```

Comment:
Nit/suggestion: You could add these as doctests at the function as this is quite useful to understand the expectations.